src/AppBundle/Entity/Dashboard.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace AppBundle\Entity;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Common\Collections\ArrayCollection;
+use Symfony\Component\Validator\Constraints as Assert;
+/**
+ * Dashboard
+ *
+ * @ORM\Table(name="sqldashboard_dashboard")
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\DashboardRepository")
+ */
+class Dashboard
```

Comment:
It's the dashboard who owns the components, I can't find this logic in the model. You need to declare the OneToMany relation in the entity.
Have a look about [doctrine mappings](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/association-mapping.html#one-to-many-bidirectional).